include/sys/libc-hooks.h

```diff
@@ -69,8 +69,13 @@ extern struct k_mem_partition z_malloc_partition;
 
 #if defined(CONFIG_NEWLIB_LIBC) || defined(CONFIG_STACK_CANARIES) || \
     defined(CONFIG_NEED_LIBC_MEM_PARTITION)
-/* Minimal libc has no globals. We do put the stack canary global in the
- * libc partition since it is not worth placing in a partition of its own.
+/* Some architectures might want to save a memory partition used by
+ * minimal libc.  In fact, many functions in minimal libs doesn't
+ * require any global variables/states. Thus we only use the partition
+ * when functions using the globals are enabled.
+ *
+ * We do put the stack canary global in the libc partition since it is
+ * not worth placing in a partition of its own.
```

Comment:
I am not sure if I understand this comment.

The scheme post-this PR is as simple as:
1. All newlib globals will be placed into `z_libc_partition`.
2. Minimal C library globals, if any, will be placed into `z_libc_partition`.
2. Stack canary globals will be placed into `z_libc_partition` since it is not worth placing in its own partition.
3. Some architectures may place the global pointer to the thread local storage in `z_libc_partition` since it is not worth placing in its own partition.

Reply:
I just copy & pasted your comment. Hope you don't mind.